src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int SIDE_LENGTH = 10;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(Figure.FIGURES_NUMBER);
+        switch (figureNumber) {
+            case 0: return new Circle(random.nextInt(SIDE_LENGTH), colorSupplier.getRandomColor());
+            case 1: return new Rectangle(random.nextInt(SIDE_LENGTH),
+                    random.nextInt(SIDE_LENGTH), colorSupplier.getRandomColor());
+            case 2: return new RightTriangle(random.nextInt(SIDE_LENGTH),
+                    random.nextInt(SIDE_LENGTH),
+                    colorSupplier.getRandomColor());
+            case 3: return new IsoscelesTrapezoid(random.nextInt(SIDE_LENGTH),
+                    random.nextInt(SIDE_LENGTH),
+                    random.nextInt(SIDE_LENGTH), colorSupplier.getRandomColor());
+            case 4: return new Square(random.nextInt(SIDE_LENGTH), colorSupplier.getRandomColor());
+            default: return this.getDefaultFigure();
```

Comment:
in this case, it is bad practice to use default, because it is not used. Let's use default instead of case 4